File: unity/Assets/Scripts/Multiplayer/Matchmaking/MatchmakingKickResponse.cs
using Michitai.Multiplayer.Errors;
using System;
using UnityEngine;

namespace Michitai.Multiplayer.Matchmaking
{
    /// <summary>
    /// Response returned when a player is successfully kicked from a matchmaking lobby.
    /// </summary>
    [Serializable]
    public class MatchmakingKickResponse : ApiResponse<EMatchmakingKickError>
    {
        /// <summary>
        /// Confirmation message for the kick operation.
        /// </summary>
        public string message;

        /// <summary>
        /// The ID of the player who was kicked.
        /// </summary>
        public int kicked_player_id;
    }
}

File: java/src/main/java/com/michitai/multiplayer/matchmaking/MatchmakingKickResponse.java
package com.michitai.multiplayer.matchmaking;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.michitai.multiplayer.ApiResponse;

/**
 * Response confirming a player was kicked from the matchmaking lobby.
 */
public class MatchmakingKickResponse extends ApiResponse {
    @JsonProperty("message")
    private String message;

    @JsonProperty("player_id")
    private int playerId;

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public int getPlayerId() {
        return playerId;
    }

    public void setPlayerId(int playerId) {
        this.playerId = playerId;
    }
}
